common/src/main/java/com/wynntils/models/horse/HorseModel.java

```diff
@@ -30,6 +31,25 @@ public Optional<HorseItem> getHorse() {
         return Models.Item.asWynnItem(McUtils.inventory().getItem(horseSlot), HorseItem.class);
     }
 
+    public Optional<CappedValue> calculateNextLevelMinutes() {
+        if (getHorse().isEmpty()) return Optional.empty();
+
+        HorseItem horseItem = getHorse().get();
+
+        if (horseItem.getLevel() == CappedValue.EMPTY || horseItem.getXp() == CappedValue.EMPTY)
+            return Optional.empty();
+        if (horseItem.getLevel().current() == horseItem.getLevel().max()) return Optional.empty();
+
+        // This is based off of a formula from https://wynncraft.wiki.gg/wiki/Horses#Levels
+        double levelProgress = (horseItem.getLevel().current() + (2f / 3f)) / 2;
```

Comment:
```suggestion
        double levelProgress = (horseItem.getLevel().current() + (2f / 3f)) / 2f;
```

Floating point conversion rules (especially in combination with division) is tricky. To be on the safe side, always mark literals as floating point.